src/routes/v3/organizations.rs

```diff
@@ -522,7 +523,105 @@ pub async fn organization_delete(
+
     let mut transaction = pool.begin().await?;
+
+    // Handle projects- every project that is in this organization needs to have its owner changed the organization owner
+    let organization_project_teams = sqlx::query!(
+        "
+        SELECT t.id FROM organizations o
+        INNER JOIN mods m ON m.organization_id = o.id
+        INNER JOIN teams t ON t.id = m.team_id
+        WHERE o.id = $1 AND $1 IS NOT NULL
+        ",
+        organization.id as database::models::ids::OrganizationId
+    )
+    .fetch_many(&mut *transaction)
+    .try_filter_map(|e| async { Ok(e.right().map(|c| crate::database::models::TeamId(c.id))) })
+    .try_collect::<Vec<_>>()
+    .await?;
+
+    // Get all project teams from the organization that do not have owner as a team_member
+    let project_teams_without_owner = sqlx::query!(
+        "
+        SELECT t.id FROM teams t
+        WHERE t.id = ANY($1) AND NOT EXISTS (
+            SELECT 1 FROM team_members tm
+            WHERE tm.team_id = t.id AND tm.user_id = $2
+        )
+        ",
+        &organization_project_teams
+            .iter()
+            .map(|x| x.0 as i64)
+            .collect::<Vec<_>>(),
+        owner_id.0
+    )
+    .fetch_many(&mut *transaction)
+    .try_filter_map(|e| async { Ok(e.right().map(|c| crate::database::models::TeamId(c.id))) })
+    .try_collect::<Vec<_>>()
+    .await?;
+
+    for project_team_without_owner in project_teams_without_owner {
+        let new_id =
+            crate::database::models::ids::generate_team_member_id(&mut transaction).await?;
+        let member = TeamMember {
+            id: new_id,
+            team_id: project_team_without_owner,
+            user_id: owner_id,
+            role: "Inherited Owner".to_string(),
+            is_owner: false,
+            permissions: ProjectPermissions::all(),
+            organization_permissions: None,
+            accepted: true,
+            payouts_split: Decimal::ZERO,
+            ordering: 0,
+        };
+        member.insert(&mut transaction).await?;
+    }
+
+    // Set is_owner to false for all team_members of projects in the organization
+    sqlx::query!(
+        "
+        UPDATE team_members tm
+        SET is_owner = FALSE
+        WHERE tm.team_id = ANY($1)
+        ",
+        &organization_project_teams
+            .iter()
+            .map(|x| x.0 as i64)
+            .collect::<Vec<_>>()
+    )
+    .execute(&mut *transaction)
+    .await?;
+
+    // Now that every project has 'owner' as a team member, we can set those projects to have the owner as the owner
+    sqlx::query!(
+        "
+        UPDATE team_members tm
+        SET is_owner = TRUE
+        WHERE tm.team_id = ANY($1) AND tm.user_id = $2
+        ",
+        &organization_project_teams
+            .iter()
+            .map(|x| x.0 as i64)
+            .collect::<Vec<_>>(),
+        owner_id.0
+    )
+    .execute(&mut *transaction)
+    .await?;
+
```

Comment:
remove this (also make it so the owner of an org can never be a team member of the project

Reply:
and also there shouldnt be an owner of a project if there is an organization